.github/workflows/ci.yml

```diff
@@ -22,131 +33,204 @@ jobs:
     needs:
       - "bandit"
       - "pydocstyle"
       - "flake8"
+      - "poetry"
       - "yamllint"
-  pylint:
+      - "black"
     runs-on: "ubuntu-22.04"
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.11"]
+        nautobot-version: ["2.0.0"]
+    env:
+      INVOKE_NAUTOBOT_CAPACITY_METRICS_PYTHON_VER: "${{ matrix.python-version }}"
+      INVOKE_NAUTOBOT_CAPACITY_METRICS_NAUTOBOT_VER: "${{ matrix.nautobot-version }}"
     steps:
       - name: "Check out repository code"
-        uses: "actions/checkout@v2"
+        uses: "actions/checkout@v4"
       - name: "Setup environment"
-        uses: "networktocode/gh-action-setup-poetry-environment@v5"
+        uses: "networktocode/gh-action-setup-poetry-environment@v4"
+      - name: "Set up Docker Buildx"
+        id: "buildx"
+        uses: "docker/setup-buildx-action@v3"
+      - name: "Build"
+        uses: "docker/build-push-action@v5"
+        with:
+          builder: "${{ steps.buildx.outputs.name }}"
+          context: "./"
+          push: false
+          load: true
+          tags: "${{ env.PLUGIN_NAME }}/nautobot:${{ matrix.nautobot-version }}-py${{ matrix.python-version }}"
+          file: "./development/Dockerfile"
+          cache-from: "type=gha,scope=${{ matrix.nautobot-version }}-py${{ matrix.python-version }}"
+          cache-to: "type=gha,scope=${{ matrix.nautobot-version }}-py${{ matrix.python-version }}"
+          build-args: |
+            NAUTOBOT_VER=${{ matrix.nautobot-version }}
+            PYTHON_VER=${{ matrix.python-version }}
       - name: "Copy credentials"
         run: "cp development/creds.example.env development/creds.env"
-      - name: "Build Container"
-        run: "poetry run invoke build"
-      - name: "Linting: Pylint"
-        run: "poetry run invoke pylint"
-    needs:
-      - "build"
+      - name: "Checking: migrations"
+        run: "poetry run invoke check-migrations"
   unittest:
+    needs:
+      - "pylint"
+      - "check-migrations"
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.8"]
-        db-backend: ["postgresql", "mysql"]
-        nautobot-version: ["2.0.0"]
-        # The include is a method to limit the amount of jobs ran. This essentially
-        # means that in addition to standard postgres and stable, also the lowest
-        # supported version and with mysql
+        python-version: ["3.8", "3.11"]
+        db-backend: ["postgresql"]
+        nautobot-version: ["stable"]
         include:
           - python-version: "3.11"
             db-backend: "postgresql"
             nautobot-version: "2.0.0"
-          - python-version: "3.11"
-            db-backend: "postgresql"
-            nautobot-version: "stable"
-          - python-version: "3.11"
-            db-backend: "mysql"
-            nautobot-version: "stable"
+          # - python-version: "3.11"
+          #   db-backend: "mysql"
+          #   nautobot-version: "stable"
```

Comment:
MYSQL unit tests are failing, in the current `develop` branch, it was using PostgreSQL tests, even when `db-backend == 'mysql'`.